File: F_Consecutive_Subsequence.py
n = int(input())
a = list(map(int, input().split()))

dp = {}
cnt = 0
x = 0

for i in range(n):
    dp[a[i]] = dp.get(a[i] - 1, 0) + 1
    if dp[a[i]] > cnt:
        cnt = dp[a[i]]
        x = a[i]

ans = []
for i in range(n - 1, -1, -1):
    if a[i] == x:
        ans.append(i + 1)
        x -= 1

print(len(ans))
print(" ".join(map(str, ans[::-1])))

File: F_Consecutive_Subsequence.cpp
#include <bits/stdc++.h>
using namespace std;

#define int long long

signed main() {
    int n, cnt = 0, x = 0;
    cin >> n;
    vector<int> a(n);
    map<int, int> dp;
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
        dp[a[i]] = dp[a[i] - 1] + 1;
        if (dp[a[i]] > cnt) {
            cnt = dp[a[i]];
            x = a[i];
        }
    }
    vector<int> ans;
    for (int i = n - 1; i >= 0; --i) {
        if (a[i] == x) {
            ans.emplace_back(i + 1);
            --x;
        }
    }
    cout << ans.size() << endl;
    for (int i = ans.size() - 1; i >= 0; --i) {
        cout << ans[i] << " ";
    }
    return 0;
}
